src/components/CreateArtist.jsx

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState, useMemo } from 'react';
+import { Deserializer } from 'jsonapi-serializer';
+import useAuth from '../hooks/useAuth';
+
+const initialValues = {
+  name: '',
+  origin: '',
+  genres: '',
+  foundedIn: '',
+  members: '',
+};
+
+export default function CreateArtist({ addArtist }) {
+  const [values, setValues] = useState(initialValues);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const { currentUser } = useAuth();
+
+  const handleSubmit = async function handleSubmit(event) {
+    setLoading(true);
+    event.preventDefault();
+    const requestOptions = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${currentUser?.access_token}`,
+      },
+      body: JSON.stringify(values),
+    };
+    fetch(`${process.env.REACT_APP_API_URL}/api/artists`, requestOptions)
+      .then((response) => {
+        if (!response.ok) {
+          setError(true);
+          return {};
+        }
```

Comment:
Esto haría que en caso de una respuesta "no ok" se devolviera `{}`, sin un error "real", lo que haría que se ejecute el siguiente `then` en la cadena, que utiliza `Deserializer` y llama a su vez a `addArtist`. O sea, estaríamos agregando un artista "vacío" a la lista de artistas.

Creo que sería mejor retornar `Promise.reject()` con algún mensaje de error, o bien, convertir todo esto a una versión con `await` (aprovechando que la función ya está marcada como async) y usar una estrategia similar al submit del login:
```suggestion
        if (!response.ok) {
          const error = await response.text();
          throw new Error(error);
        }
```

Reply:
No me permite utilizar el `await`, será porque está dentro del `fetch`?

Reply:
Claro, por eso decía que tendrías que convertir todo esto a una versión con `await`, desde `fetch`. Por eso tal vez sea más simple algo como:
```javascript
setError(true);
return Promise.reject('Something went wrong').
```
Lo único es que perderíamos el mensaje que viene desde la API. Alternativamente, se podría hacer algo como:
```javascript
setError(true);
return response.text().then((error) => Promise.reject('Something went wrong')).
```
No es lo más ordenado, pero permitiría acceder al error real desde la API, y que luego esté disponible en el `catch` siguiente en la cadena.

Reply:
Ya que eslint solo deja rechazar por errores, quedaría así:

`return response.text().then((message) => Promise.reject(new Error(message)));`

(error como nombre de argumento era shadowname al error mismo)

Reply:
Yo creo que podemos dejarlo así. Debería entenderse.